File: src/main/java/com/geopark/web/model/entity/Naturallandscape.java
package com.geopark.web.model.entity;

import com.alibaba.excel.annotation.ExcelIgnore;
import com.alibaba.excel.annotation.ExcelProperty;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * <p>
 * 自然景观
 * </p>
 *
 * @author lichee
 * @since 2019-09-19
 */
@Data
@EqualsAndHashCode(callSuper = false)
@TableName("naturallandscape")
public class Naturallandscape implements Serializable {

    @ExcelIgnore
    private static final long serialVersionUID = 1L;

    @TableId(value = "id", type = IdType.AUTO)
    @ExcelIgnore
    private Integer id;

    /**
     * 地质公园ID
     */
    @ExcelProperty(value = "地质公园ID",index = 0)
    private Integer parkId;

    /**
     * 自然景观编号
     */
    @ExcelProperty(value = "自然景观编号",index = 1)
    private String naturalnumber;

    /**
     * 景观名称
     */
    @ExcelProperty(value = "景观名称",index = 2)
    private String nname;

    /**
     * 地理位置
     */
    @ExcelProperty(value = "地理位置",index = 3)
    private String position;

    /**
     * 交通状况
     */
    @ExcelProperty(value = "交通状况",index = 4)
    private String traffic;

    /**
     * 经度
     */
    @ExcelProperty(value = "经度",index = 5)
    private String lat;

    /**
     * 纬度
     */
    @ExcelProperty(value = "纬度",index = 6)
    private String lng;

    /**
     * 海拔高度
     */
    @ExcelProperty(value = "海拔高度",index = 7)
    private String altitude;

    /**
     * 景观特色
     */
    @ExcelProperty(value = "景观特色",index = 8)
    private String feature;

    /**
     * 文物保护单位
     */
    @ExcelProperty(value = "文物保护单位",index = 9)
    private String level;

    /**
     * 批准时间
     */
    @ExcelProperty(value = "批准时间",index = 10)
    private String approvedtime;

    /**
     * 保护现状
     */
    @ExcelProperty(value = "保护现状",index = 11)
    private String status;

    /**
     * 照片
     */
    @ExcelProperty(value = "照片",index = 12)
    private String img;

    @ExcelIgnore
    private LocalDateTime createdAt;
    @ExcelIgnore
    private LocalDateTime updatedAt;
    @ExcelIgnore
    private LocalDateTime deletedAt;


}

File: src/main/resources/static/js/pkmanage/naturallandscape.js
//公园概况

layui.use(['config', 'lichee', 'jquery', 'layer', 'table', 'form', 'laydate','imageUpload'], function () {

    var config = layui.config;
    var lichee = layui.lichee;
    var $ = layui.jquery;
    var layer = layui.layer;
    var table = layui.table;
    var form = layui.form;
    var laydate = layui.laydate;
    var imageUpload = layui.imageUpload;

    var tableInfo = table.render({
        elem: '#table',
        toolbar: '#toolbar',
        defaultToolbar: [],
        //width:'auto',
        url: '/naturallandscape/page',
        title: '自然景观',
        page: true,
        headers: {Authorization: config.getToken()},
        request: config.request,
        parseData: config.parseData,
        response: config.response,
        cols: [[
            //{type: 'checkbox', fixed: 'left'},
            // {field: 'id', title: 'ID'},
            {field: 'parkId', align: 'center', sort: true, title: '地质公园ID'},
            {field: 'naturalnumber', align: 'center', sort: true, title: '景观编号'},
            {field: 'nname', align: 'center', sort: true, title: '景观名称'},
            {field: 'position', align: 'center', sort: true, title: '地理位置'},
            //{field: 'traffic', align: 'center', sort: true, title: '交通状况'},
            {field: 'lat', align: 'center', sort: true, title: '经度'},
            {field: 'lng', align: 'center', sort: true, title: '纬度'},
            //{field: 'altitude', align: 'center', sort: true, title: '海拔高度'},
            //{field: 'feature', align: 'center', sort: true, title: '特色'},
            //{field: 'level', align: 'center', sort: true, title: '保护单位'},
            //{field: 'approvedtime', align: 'center', sort: true, title: '批准时间'},
            //{field: 'status', align: 'center', sort: true, title: '保护现状'},
            {fixed: 'right',align : 'center', title: '操作', toolbar: '#bar', width: 180}
        ]]
    });

    table.on('toolbar(tableFilter)', function (obj) {
        var checkStatus = table.checkStatus(obj.config.id);
        switch (obj.event) {
            case 'add':
                showEditModel();
                break;
            case 'getCheckLength':
                var data = checkStatus.data;
                layer.msg('选中了：' + data.length + ' 个');
                break;
            case 'isAll':
                layer.msg(checkStatus.isAll ? '全选' : '未全选');
                break;
        };
    });

    table.on('tool(tableFilter)', function (obj) {
        var data = obj.data;
        if (obj.event === 'del') {
            layer.confirm('确定要删除吗？', function (index) {
                layer.load(2);
                lichee.delete('/naturallandscape/' + obj.data.id, {}, function () {
                    layer.closeAll('loading');
                    layer.msg('删除成功', {icon: 1});
                    obj.del();
                });
            });
        } else if (obj.event === 'edit') {
            showEditModel(data);
        }
    });

    var showEditModel = function (data) {
        layer.open({
            title: data ? '详情/修改' : '添加',
            type: 1,
            area: ['800px', '600px'],
            content: $('#form-model').html(),
            success: function () {
                $('#form')[0].reset();
                lichee.get('/park/list', {async: false}, function (data) {
                    $('#parkId').vm({parks: data.result});
                    form.render('select');
                });
                imageUpload.init("uploadBtn", "img", "humanlandscape");
                if (data) {
                    form.val('formFilter', data);
                    imageUpload.initImageList(data.img);
                }
                $('#form .close').click(function () {
                    layer.closeAll('page');
                });
            }
        });
    };

    var form = layui.form;
    form.on('submit(formSubmit)', function (data) {
        layer.load(2);
        if (data.field.id) {
            lichee.put('/naturallandscape/' + data.field.id, {data: data.field}, function (res) {
                callFunction(res);
            });
        } else {
            lichee.post('/naturallandscape', {data: data.field}, function (res) {
                callFunction(res);
            });
        }
        return false;
    });

    form.verify({
        commonLength: [/^[\S]{1,100}$/, '字符长度最大100'],
        descLength: [/^[\S]{0,180}$/, '字符长度最大180']
    });

    var callFunction = function (res) {
        layer.closeAll();
        if (res.status == 200) {
            layer.msg("修改成功",{icon: 1});
            tableInfo.reload({where: lichee.getSearchForm()});
        } else if(res.status == 201)  {
            layer.msg("添加成功",{icon: 1});
            tableInfo.reload({where: lichee.getSearchForm()});
        }
    };

    $('#searchButton').click(function () {
        tableInfo.reload({where: lichee.getSearchForm()});
    });

    laydate.render({
        elem: '#startTime',
        type: 'datetime',
        value: '',
        trigger: 'click'
    });

    $(".laydate").each(function (i,e) {
        laydate.render({
            elem: this,
            type: 'datetime',
            value: '',
            trigger: 'click'
        });
    });

});